### PR TITLE
Fixed PHP 8.1 compatibility

### PR DESCRIPTION
**Pull request by lamka02sk**
Hi, there were some PHP type errors in 8.1, so I fixed them. There is no guarantee that more of these won't pop out in the future, but this is what we found during testing for now.

### CODE REVIEW

**Review by tsantos84 — DISMISSED** [review]
Thanks for fixing it.

File: src/ApmLogger.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ private function getLevelValue(string $level): int
     /**
      * @inheritDoc
      */
-    public function log($level, $message, array $context = array())
+    public function log($level, string|\Stringable $message, array $context = []) : void
```
Comment (dstepe) [inline src/ApmLogger.php:48]: The union type and use of `Stringable` will force the minimum supported version of PHP to PHP 8.0. That will be a major change and require a bit more consideration.

**Review by dstepe — COMMENTED** [review]
Ensuring PHP >8.0 compatibility is overdue and I appreciate the work you are doing. I think a first step will be to add 8.0 and 8.1 to the github workflow matrix and see where we stand. If we can reach compatibility with >8.0 without introducing breaking changes, then we can decide what type of release makes sense. If, however, 8.0+ compatibility cannot be achieved while maintaining <8.0 compatibility, we'll need to think hard about how to proceed.

**Comment by dstepe** [comment]
@lamka02sk Can you provide more context for the changes you are proposing? I just completed a quick test of the latest version of this package (v7.5.3) under PHP 8.1.2 and see no reported issues. I have also updated the GitHub actions matrix to run linting and unit tests under PHP 8.0 and 8.1 (see https://github.com/nipwaayoni/elastic-apm-php-agent/actions/runs/1790344290). Those also execute without reporting any issues.

I absolutely want this package to be compatible with PHP 8.1, but I'm not ready to require PHP >=8.0 yet. What problem are you seeing that you want to solve with the PR?

**Comment by lamka02sk** [comment]
@dstepe I looked more into the issue and found some strange behaviour in our composer installation. The problem was that we had package psr/log installed in the latest version 3.0 (debugbar package defaults to 3.0) which contains multiple changes in method declarations, causing the errors. Locking the package to version 1.1 solved the issues. I am closing this PR now, sorry for making trouble.

**Comment by dstepe** [comment]
@lamka02sk I'm glad you got it worked out. No trouble at all. It's nice to know the package is used.

**Pull request closed without merging**